coderd/httpapi/search.go

```diff
@@ -0,0 +1,73 @@
+package httpapi
+
+import (
+	"strings"
+
+	"golang.org/x/xerrors"
+)
+
+// WorkspaceSearchQuery takes a query string and breaks it into it's queryparams
+// as a set of key=value.
+func WorkspaceSearchQuery(query string) (map[string]string, error) {
```

Comment:
Maybe I should move this to `workspaces.go`? 

I kinda want to make this more generic for when we add a `UserSearchQuery`, but felt I'll make it generic when we get that far.

Reply:
Moved. Might move again if we make other queries for users/templates page.